### DISCUSSION TITLE
Grid Items to Print

### DISCUSSION THREAD

**Discussion by jbk68** [cortexrd/Knack-Toolkit-Library#187] [General]
I was just wondering if KTL currently had the ability to choose what fields are printed using Knack's "print" button found in the header.

If it is currently available, could someone please let me know how?  If it's not, it would be a nice feature that a developer of an app could choose what fields will be printed.  

Right now, when a user press print, all fields including action links/buttons are printed. 

Any and all current solutions would are appreciated.

Thank you... JON

**Comment by amandajane-mo** [comment]
Hi Jon,

Not a current ktl feature, we use some css and _cls for this. I’ll send you
over what we use tomorrow.

Amanda

**Comment by jbk68** [comment]
Thank you Amanda...

**Comment by amandajane-mo** [comment]
Hi Jon,

We use css like this to hide certain elements from print, things we want to always hide (like buttons, action links in tables etc.) are listed in there and we also have a class hide-on-print, which we add to elements we want to hide to hide using _cls

```
@media print{
  /* Don't display following when printing */
  .hide-on-print, body .kn-sig-clear, body input[type="button"], body .kn-required, i.fa.fa-edit, .kn-menu, #appUsersMessage, .kn-table-link { 
    display: none!important;
  }
}
```

So to hide specific columns on a grid we use:

`_cls=[hide-on-print], [ktlTarget, $('.field_3779, .field_5380')]`

to hide field 3779 and 5380 from print.

Having this as field and/or view keyword would be great.

Hope that helps,

Amanda

**Comment by jbk68** [comment]
Hey Amanda, I really appreciate all your help and will give this a try.

Thank you... JON

**Comment by jbk68** [comment]
Hey Amanda, have you guys figured out how to stop certain field and their data from being exported using Knack's export feature?

I have a lot of hidden fields that contain unnecessary data (ie text formulas, calculation etc...) that are hidden using _uc but I can't seem to figure out how to NOT make these fields from being exported.  Sure would be nice if there was a way to choose what fields are exported and which ones are not.

Thanks... JON
